csrc/fusion.h

```diff
@@ -105,30 +107,10 @@ struct AliasInfo {
   bool operator!=(const AliasInfo& other) const {
     return !(*this == other);
   }
-
-  std::string toString() const {
```

Comment:
What's the motivation for switching from `toString` to registering `ostream` operators? The latter may reduce some verbosity from error messages. I prefer having `toString` method for gdb-ing. 
Do the struct specific `ostream` methods work reliably for you -- I have had some issues with that?